openprovider/apis/openprovider_api.php

```diff
@@ -99,12 +99,14 @@ public function call(string $cmd, array $args = []): Response
             $service->getConfig()->setAccessToken($this->api_config->getToken());
         }
 
+        $modifiedArgs = $this->params_creator->modifyArgsIfDomainIdn($args);
+
         $this->last_request = new LastRequest();
-        $this->last_request->setArgs($args);
+        $this->last_request->setArgs($modifiedArgs);
```

Comment:
I think it's better to create separate class for each command for which you build request parameters, e.g.
params_builder/create_domain.php
params_builder/search_domains.php
etc.
and here you will build needed class like
$creator = ParamsCreatorFactory->build($cmd);
$params = $creator->createParameters($args, $service, $apiMethod);

Each params creator will do something like
CreateDomainParamsCreator extends ParamsCreator {
public function createParameters(array $args, $client, string $method): array {
 return partent::createParameters($this->idnToAscii($args), $client, $method);
}